parser.go

```diff
@@ -944,18 +948,59 @@ func (p *parser) parse(prec precedence) (astNode, error) {
 			op := p.pop()
 			switch op.kind {
 			case tokenBracketL:
-				index, err := p.parse(maxPrecedence)
-				if err != nil {
-					return nil, err
+				// handle slice
+				var indexes [3]astNode
+				which := 0
+
+				var end *token
+				readyForNextIndex := true
+				for which < 3 {
+					if p.peek().kind == tokenBracketR {
+						end = p.pop()
+						break
+					} else if p.peek().data == ":" {
+						which++
+						end = p.pop()
+						readyForNextIndex = true
+					} else if p.peek().data == "::" {
+						which += 2
+						end = p.pop()
+						readyForNextIndex = true
+					} else if readyForNextIndex {
+						indexes[which], err = p.parse(maxPrecedence)
+						if err != nil {
+							return nil, err
+						}
+						readyForNextIndex = false
+					} else {
+						_, err := p.popExpect(tokenBracketR)
```

Comment:
I'm missing something here, shouldn't the fact that the first `if p.peek().kind == tokenBracketR` failed mean that this `p.popExpect(tokenBracketR)` will always fail?